apps/actions-dependencies-updater/src/workflows.mts

```diff
@@ -0,0 +1,401 @@
+   */
+  jobs: Job[];
+}
+
+interface Job {
+  name: string;
+
+  containingWorkflow: string;
+
+  directDependencies: Action[];
+
+  indirectDependencies?: Action[];
+}
+
+/**
+ * Parses all workflows in the repository and returns them as a map of workflow name to workflow object.
+ * @param ctx the RunContext
+ * @returns a map of workflow name to workflow object, containing all workflows in the repository
+ */
+export async function parseWorkflows(ctx: RunContext) {
+  const workflowDirectory = join(ctx.repoDir, ".github", "workflows");
+  const workflowFiles = await listAllYamlFiles(workflowDirectory);
+
+  const workflows: WorkflowByName = {};
+
+  for (const workflowFilePath of workflowFiles) {
+    const workflow = await parseWorkflow(ctx, workflowFilePath);
+    workflows[workflow.file] = workflow;
+  }
+
+  return workflows;
+}
+
+/**
+ * Parses a workflow file and returns the workflow object
+ * @param ctx the RunContext
+ * @param workflow the name of the workflow
+ * @param path the path to the workflow file
+ * @returns the parsed workflow
+ */
+async function parseWorkflow(ctx: RunContext, path: string): Promise<Workflow> {
+  const file = basename(path);
+
+  log.debug(`Processing workflow: ${file}`);
+
+  const fileStr = await readFile(path, "utf-8");
+  const parsedFile = YAML.parse(fileStr) as WorkflowSchema;
+
+  if (parsedFile == null) {
+    log.warn(`Parsed as empty workflow file: ${file}`);
+    return { file, path, jobs: [] };
+  }
+
+  const { name, jobs: jobDefinitions } = parsedFile;
+
+  if (!jobDefinitions) {
+    log.warn(`No jobs found in workflow file: ${file}`);
+    return { name, file, path, jobs: [] };
+  }
+
+  const jobsPromises = Object.entries(jobDefinitions).map(
+    async ([jobKey, jobDefinition]) =>
+      parseJob(ctx, file, jobKey, jobDefinition),
+  );
+
+  const jobs = await Promise.all(jobsPromises);
+  return { name, file, path, jobs };
+}
+
+/**
+ * Parses a single job definition contained within a workflow.
+ * @param ctx the RunContext
+ * @param workflow the name of the workflow
+ * @param jobKey the key of the job
+ * @param jobDefinition the job definition
+ * @returns the parsed job
+ */
+async function parseJob(
+  ctx: RunContext,
+  workflow: string,
+  jobKey: string,
+  jobDefinition: WorkflowSchema["jobs"]["key"],
+): Promise<Job> {
+  log.debug(`Processing Job: ${workflow} - ${jobKey}`);
+
+  if ("uses" in jobDefinition) {
+    log.warn(
+      "Found Job which is uses a workflow_call, to initiate another workflow. Parsing this is not directly supported, but if the workflow is in the same repository it should be parsed regularly.",
+    );
+    return {
+      name: jobKey,
+      containingWorkflow: workflow,
+      directDependencies: [],
+    };
+  }
+
+  if (!("steps" in jobDefinition)) {
+    log.warn(
+      "No steps found in job definition, bad job definition? Assuming no dependencies",
+    );
```

Comment:
This should error, no steps creates an invalid workflow

Reply:
This script doesn't care. If there's no steps then no problem, there's no updates.